internal/pkg/tracer/tracer.go

```diff
@@ -33,33 +41,55 @@ func FromContext(ctx context.Context) trace.Tracer {
 // Backend is an abstraction for tracng backend (Datadog, Sentry, ...)
 type Backend interface {
 	Init(componentName string) error
+	Type() config.TracerProvider
+	Capture(*datastore.Project, string, *net.Response, time.Duration)
+	Shutdown(ctx context.Context) error
 }
 
-type shutdownFn func(context.Context) error
+type NoOpBackend struct{}
+
+func (NoOpBackend) Init(componentName string) error { return nil }
+func (NoOpBackend) Type() config.TracerProvider {
+	return ""
+}
+func (NoOpBackend) Capture(*datastore.Project, string, *net.Response, time.Duration) {
 
-func noopShutdownFn(context.Context) error {
+}
+func (NoOpBackend) Shutdown(context.Context) error {
 	return nil
 }
 
 // Global tracer Init function
-func Init(tCfg config.TracerConfiguration, componentName string) (shutdownFn, error) {
+func Init(tCfg config.TracerConfiguration, componentName string, licenser license.Licenser) (Backend, error) {
 	switch tCfg.Type {
 	case config.SentryTracerProvider:
+		st := NewSentryTracer(tCfg.Sentry)
 		if tCfg.Sentry == (config.SentryConfiguration{}) {
-			return noopShutdownFn, ErrInvalidTracerConfiguration
+			return st, ErrInvalidTracerConfiguration
+		}
+
+		return st, st.Init(componentName)
+
+	case config.DatadogTracerProvider:
+		dt := NewDatadogTracer(tCfg.Datadog, licenser)
+		if !licenser.DatadogTracing() {
+			log.Error(ErrTracerFeatureUnavailable.Error())
```

Comment:
We should pass the log instance so this line respect the configured log level

Reply:
Noted